frlearn/uncategorised/fuzzy_quantifiers.py

```diff
@@ -0,0 +1,104 @@
+"""
+    Some fuzzy quantifiers.
+    All the following functions are the implementations of defintions of some fuzzy quantifiers in [1].
+
+    References
+    ----------
+    [1] THEERENS, Adnan et CORNELIS, Chris.
+    Fuzzy rough sets based on fuzzy quantification.
+    Fuzzy Sets and Systems, 2023, vol. 473, p. 108704.
+"""
+
+
+def choquet_integral(f, mu):
+    """
+        Implementation of the Choquet_integral
+        parameters:
+        ----------
+        f: list
+           Define a list that contains the values of the function
+        mu: list
+            Define a list that contains the values of the monotone measure.
+    """
+    n = len(mu)
+    f_sorted = sorted(f)
+    integral = 0
+    for i in range(n):
+        if i == 0:
+            integral = mu[0] * f_sorted[0]
+        else:
+            integral += mu[i] * (f_sorted[i] - f_sorted[i - 1])
+    return integral
+
+
+def unary_zadeh_model(rim, A):
+    """
+    Evaluation of unary sentences of the form “Λ X’s are A’s” in Zadeh’s model, where Λ is a RIM quantifier
+    parameters:
+    ----------
+    rim: function
+        Definition of the RIM quantifier
+    A: table (list)
+       List of value of the Fuzzy set
+    """
+
+    return rim(sum(A) / len(A))
+
+
+def binary_zadeh_model(rim, A, B):
+    """
+    Evaluation of binary sentences of the form “Λ A’s are B’s” in Zadeh’s model, where Λ is a RIM quantifier
+    parameters:
+    ----------
+    rim: function
+         Definition of the RIM quantifier
+    A, B: table (list)
+          List of value of the Fuzzy sets
+    """
+    return rim(sum([el for el in A if el in B]) / sum(A))
```

Comment:
The assumption is (should be) that the values of `A` and `B` correspond to each other (i.e. to the elements of the underlying set). Therefore, I think this should be `rim(sum([min(a, b) for a, b in zip(A, B)]) / sum(A))`

Reply:
you're absolutely right